Sources/CombineSchedulers/TestScheduler.swift

```diff
@@ -118,8 +121,8 @@
           self.lock.unlock()
           return
         }
-
-        self.now = next.date
+        
+        self.now = next.date.advanced(by: minimumTolerance)
```

Comment:
I think my hunch was that a call to `scheduler.advance()` would incur the minimum tolerance and no more, but it sounds like this will incur the minimum tolerance for every work item, which seems a little less easy to understand from the outside, and more prone to causing test failures when your code changes. What do you think about calling it a single time instead of in the loop?

Reply:
The idea was to replicate in a predictable way, how schedulers work in real life. If I set a timer for 1 second, it's never just exactly 1 second. There is always an additional bit of time albeit very small. But it can add up quite quickly if not accounted for.

If there are several units of work scheduled, you would want them all to be triggered at their scheduled date plus the the delay which you specified when you create the scheduler. 

Perhaps it's confusing using the minimum tolerance for this.

Reply:
Another thought on this. 

As the motivation for this is to replicate the slight difference between the scheduled time and the time the action actually takes place, it might make more sense to update the schedule date directly as the action is scheduled. 

This way it's easier to reason. You know if you have a test scheduler with a drift value of 0.05 and you schedule some work at 1 second, that it will occur once the scheduler advances 1.05 seconds.

The reason you might want this is if you are performing repeated scheduling and precision matters to you. If you know what time an action was scheduled for you can take the difference between the scheduled time and the actual time, and offset the next scheduled event.